repo/api_server_repository.go

```diff
@@ -346,6 +355,13 @@ func openRestAPIRepository(ctx context.Context, si *APIServerInfo, password stri
 
 	rr.omgr = omgr
 
+	par.registerEarlyCloseFunc(func(ctx context.Context) error {
+		// release debugging resources as part of early debug.  This leaves the debugger
+		// a chance to use runtime resources before they too are released
+		rr.CloseDebug(ctx)
+		return nil
+	})
+
```

Comment:
since `openRestAPIRepository()` calls `debug.StartProfileBuffers(ctx)`, I think it'd be okay to stop the profile buffers using `debug.StopProfileBuffers(ctx)`, that'd keep the layering consistent.